recipes/sisl/build.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/sh
+
+# There are some issues in the linking on Py3 vs Py2
+# Hence we need to figure out which version we are using
+# to fix the f2py linking step.
+_V=${CONDA_PY:0:1}
+_v=${CONDA_PY:1:2}
+
+if [ -z "$MACOSX_DEPLOYMENT_TARGET" ]; then
+    export LDFLAGS="-shared $LDFLAGS"
+else
+    if [ $_V -eq 3 ]; then
+	export LDFLAGS="-shared $LDFLAGS -lpython${_V}.${_v}m"
```

Comment:
I think this is better, since the `m` suffix is optional and it's conceivable it'd drop or change in the future:
```
$(ldd $PYTHON | grep 'libpython.*\.so' | cut -d' ' -f3)
```
The Mac equivalent, though for whatever reason I guess you don't really need it, [might be](https://github.com/conda-forge/shogun-feedstock/blob/master/recipe/meta.yaml#L52)
```
$(otool -L $PYTHON | grep 'libpython.*\.dylib' | tr '\t' ' ' | cut -d' ' -f2 | sed "s|@rpath|$PREFIX/lib|")
```
Also relevant: https://github.com/conda/conda-build/issues/2130.

Reply:
Oh, those give the full path to the library (because I was doing that for cmake before). I guess it wants an extra, say, `| xargs basename | sed -e 's/lib/-l/; s/\.so.*//'`

Reply:
Thanks, trying a few things in the commit